src/mapping-v3-core.ts

```diff
@@ -892,6 +919,225 @@ function _handleOwnershipTransferred<T>(
   refreshContract(contract, event.block.timestamp);
 }
 
+export function handleExternalAssetDependencyUpdated(
+  event: ExternalAssetDependencyUpdated
+): void {
+  const engineFlexContract = getV3EngineFlexContract(event.address);
+  if (engineFlexContract) {
+    _handleExternalAssetDependencyUpdated(engineFlexContract, event);
+    return;
+  }
+  log.warning("[WARN] Unknown V3 coreType for contract at address {}.", [
+    event.address.toHexString()
+  ]);
+}
+
+function _handleExternalAssetDependencyUpdated<T>(
+  contract: T,
+  event: ExternalAssetDependencyUpdated
+): void {
+  if (!(contract instanceof GenArt721CoreV3_Engine_Flex)) {
+    return;
+  }
+  let project = Project.load(
+    generateContractSpecificId(event.address, event.params._projectId)
+  );
+
+  if (!project) {
+    log.warning(
+      "Project not found for ExternalAssetDependencyUpdated event",
+      []
+    );
+    return;
+  }
+
+  const assetEntity = new ProjectExternalAssetDependency(
+    generateProjectExternalAssetDependencyId(
+      project.id,
+      event.params._index.toString()
+    )
+  );
+  assetEntity.cid = event.params._cid;
+  assetEntity.project = project.id;
+  assetEntity.index = event.params._index;
+  assetEntity.dependencyType =
+    FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[event.params._dependencyType];
+
+  if (assetEntity.dependencyType === "ONCHAIN") {
+    const projextExternalAssetDependency = contract.try_projectExternalAssetDependencyByIndex(
+      event.params._projectId,
+      event.params._index
+    );
+    assetEntity.bytecodeAddress =
+      projextExternalAssetDependency.value.bytecodeAddress;
+    assetEntity.data = projextExternalAssetDependency.value.data;
```

Comment:
I'm fine with adding the `.reverted` check personally (and think it will be easier at this point). but also, thinking more on it we may just be able to use a regular contract call vs 'try_'. The only way this contract call reverts is if the external asset dependency, which we just received an event for in this handler, wouldn't exist (and this shouldn't be possible).

Reply:
Agree, and we certainly do not rely on checking for reverted in other handlers. At some point we should codify which is preferred in this repo, but for now seems like we either remove the `try_`, or add the `!reverted` check?

Reply:
went with removing `try_` 👍